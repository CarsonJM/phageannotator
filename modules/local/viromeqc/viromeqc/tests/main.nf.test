nextflow_process {

    name "Test Process: VIROMEQC_VIROMEQC"
    script "../main.nf"
    process "VIROMEQC_VIROMEQC"
    tag "modules"
    tag "modules_local"
    tag "viromeqc/viromeqc"
    tag "viromeqc/install"

    test("sarscov2 - illumina - test_1_fastq_gz") {
        setup {
            run("VIROMEQC_INSTALL") {
                script "../../install/main.nf"
            }
        }

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    [
                        file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkifExists: true),
                        file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkifExists: true),
                    ]
                ]
                input[1] = VIROMEQC_INSTALL.out.viromeqc_index
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
