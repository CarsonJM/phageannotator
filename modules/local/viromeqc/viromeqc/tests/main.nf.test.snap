{
    "sarscov2 - illumina - test_1_fastq_gz": {
        "content": [
            {
                "0": [
                    "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                ],
                "1": [
                    "versions.yml:md5,1d793f1196d60e1c94969fb0a103e353"
                ],
                "enrichment": [
                    "test.viromeqc.tsv:md5,6acbffc97f75774cf9fcc953b98913ed"
                ],
                "versions": [
                    "versions.yml:md5,1d793f1196d60e1c94969fb0a103e353"
                ]
            }
        ],
        "timestamp": "2023-12-28T13:22:47.999767324"
    }
}
